### DISCUSSION TITLE
How to add rabbitmq lazy queue with ruby code?

### DISCUSSION THREAD

**Discussion by bhaskarpc** [rabbitmq/rabbitmq-server#5178] [Other] [answered]
Hi All,

We are trying to create a lazy queue using ruby code but we are not able to. We are using this code

rmq_channel.queue("test_queue", {:durable => true, :arguments => {"x-queue-mode" => "lazy"}})

but the above code is adding Args in the features section and showing x-queue-mode": "lazy"

please check the below image for reference [here](https://i.stack.imgur.com/1jheK.png)

We want to create the lazy queue something like this [please check this](https://i.stack.imgur.com/PfYzS.png)

Can someone please help us here how to create the lazy queues using ruby code?

Thanks

**Comment by michaelklishin** [comment, marked as answer]
What leads you to believe that your queue is "not lazy"? The Bunny code line is correct.

The recommended way is to [use a policy](https://rabbitmq.com/lazy-queues.html#configuration).

3.10 users should arguably use the v2 classic queue storage for new clusters instead. It's a node-wide `rabbitmq.conf` setting:

``` ini
classic_queue.default_version = 2
```
